sync/syncer.go

```diff
@@ -289,63 +284,65 @@ func (s *Syncer) getLayerHashes(index types.LayerID) (map[string]p2p.Peer, error
 		return nil, errors.New("no peers")
 	}
 	// request hash from all
-	wg := sync.WaitGroup{}
 	ch := make(chan *peerHashPair, len(peers))
+	kill := make(chan struct{})
 
-	resCounter := len(peers)
-	for _, p := range peers {
-		c, err := s.sendLayerHashRequest(p, index)
+	for _, peer := range peers {
```

Comment:
this looks quite similar to the getIdsForHashes func maybe they can share some code.